modules/spring-config/src/main/java/org/mule/runtime/config/internal/DependencyGraphLifecycleObjectSorter.java

```diff
@@ -175,25 +175,25 @@ public int compare(BeanWrapper o1, BeanWrapper o2) {
   }
 
   /**
-   * Provides the information that will be needed for the comparison during the top sort
-   * 
+   * Provides the information that will be needed for the comparison while sorting objects. This information will be used only for
+   * the tie-breaking in cases of partial ordering.
+   *
    * @param lookupObjectsForLifecycle lifecycle object list which is ordered based on the type
    */
-  public void setLifeCycleObjectNameOrderMap(Map<String, Object> lookupObjectsForLifecycle) {
-    int index = 0;
+  public void setLifeCycleObjectList(Map<String, Object> lookupObjectsForLifecycle) {
     for (Map.Entry<String, Object> entry : lookupObjectsForLifecycle.entrySet()) {
       String objectName = entry.getKey();
-      lifecycleObjectNameOrderMap.put(objectName, index++);
+      lifecycleObjectList.add(objectName);
```

Comment:
Here we should convert from `List` of ordered components to `Map` from name to position (priority). What we were doing before was OK, but instead of iterating the `entrySet` we would be iterating the list that comes as parameter.